src/components/shared/formSignIn/FormSignIn.tsx

```diff
@@ -0,0 +1,97 @@
+import { FC, useState, useEffect } from 'react';
+
+import { TextInput, Button } from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import { UserRole } from 'api/types';
+import { delay } from 'components/layout/signIn/config';
+import {
+  minWordLength,
+  placeholderTextInput,
+  sizeButton,
+  sizeTextInput,
+} from 'components/shared/formSignIn/config';
+import { InputValidation, TypeButton } from 'components/shared/formSignIn/types';
+import useDebounced from 'hooks/debounced/useDebounced';
+import { useAppDispatch, RootState } from 'store/store';
+import { userThunk } from 'store/user/userData.api';
+
+const FormSignIn: FC = () => {
+  const dispatch = useAppDispatch();
+  const user = useSelector((state: RootState) => state.user.user);
+  const isNameAvailable = useSelector((state: RootState) => state.user.isNameAvailable);
+  const [nickname, setNickname] = useState('');
+  const navigate = useNavigate();
+
+  const form = useForm({
+    initialValues: { name: '' },
+    validateInputOnChange: true,
+    validate: {
+      name: (value: string) => {
+        if (value.length < minWordLength) {
+          return InputValidation.Length;
+        }
+
+        setNickname(value);
+
+        return null;
+      },
+    },
+  });
+
+  const checkEnterName = async (): Promise<void> => {
+    if (nickname !== '') {
+      dispatch(userThunk.checkFieldName(nickname));
+    }
+  };
+
+  useDebounced({
+    callback: () => checkEnterName(),
+    delay,
+    dependencies: [nickname],
+  });
+
+  useEffect(() => {
+    if (isNameAvailable) return;
+
+    form.setFieldError('name', InputValidation.Invalid);
+  }, [isNameAvailable]);
+
+  const setUser = async (): Promise<void> => {
```

Comment:
I thought that if I call an asynchronous function inside a function, I should specify async. But I was wrong. If the function does not return anything, you do not need to specify async because this function is not asynchronous. Am I right?